samples/sqltransport/store-and-forward/Version_6/Sender/SendThroughLocalQueueBehavior.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using NServiceBus.DelayedDelivery;
+using NServiceBus.DeliveryConstraints;
+using NServiceBus.Extensibility;
+using NServiceBus.ObjectBuilder;
+using NServiceBus.Pipeline;
+using NServiceBus.Routing;
+using NServiceBus.Transports;
+
+
+class SendThroughLocalQueueRoutingToDispatchConnector : ForkConnector<IRoutingContext, IDispatchContext>
+{
+    public SendThroughLocalQueueRoutingToDispatchConnector(string localAddress)
+    {
+        this.localAddress = localAddress;
+    }
+
+    public override Task Invoke(IRoutingContext context, Func<Task> next, Func<IDispatchContext, Task> fork)
+    {
+        IncomingMessage incomingMessage;
+        bool fromHandler = context.Extensions.TryGet(out incomingMessage);
+
+        DelayedDeliveryConstraint constraint;
+        if (context.Extensions.TryGetDeliveryConstraint(out constraint) || fromHandler)
```

Comment:
When it comes from a handler, it means it already was/is in the local incoming queue so no need to route outgoing messages through it. If the destination is offline, the message will be retried by means of FLR/SLR.